crates/router/src/core/routing.rs

```diff
@@ -640,7 +640,75 @@ pub async fn unlink_routing_config(
     }
 }
 
-//feature update
+#[cfg(all(
+    feature = "v2",
+    feature = "routing_v2",
+    feature = "business_profile_v2"
+))]
+pub async fn update_default_fallback_routing(
+    state: SessionState,
+    merchant_account: domain::MerchantAccount,
+    key_store: domain::MerchantKeyStore,
+    profile_id: String,
+    updated_config: Vec<routing_types::RoutableConnectorChoice>,
+) -> RouterResponse<Vec<routing_types::RoutableConnectorChoice>> {
+    metrics::ROUTING_UPDATE_CONFIG.add(&metrics::CONTEXT, 1, &[]);
+    let db = state.store.as_ref();
+    let key_manager_state = &(&state).into();
+    let profile = core_utils::validate_and_get_business_profile(
+        db,
+        key_manager_state,
+        &key_store,
+        Some(&profile_id),
+        merchant_account.get_id(),
+    )
+    .await?
+    .get_required_value("BusinessProfile")?;
+    let profile_wrapper = admin::BusinessProfileWrapper::new(profile);
+    let default_config = profile_wrapper.get_default_fallback_list_of_connector_under_profile()?;
+
+    utils::when(default_config.len() != updated_config.len(), || {
```

Comment:
se more meaningful words(should have explanation about underlying data) for variable names, default_list_of_connectors and updated_list_of_connectors